### PR TITLE
DRILL-4786: Read the metadata cache file from the least common ancest…

### PR DESCRIPTION
**Pull request by amansinha100**
…or directory when multiple partitions are selected.

Handle wildcards appropriately when metadata cache is present.

### CODE REVIEW

**Comment by amansinha100** [comment]
@jinfengni could you pls review the PR since you reviewed the related PR earlier ?  thanks.

File: exec/java-exec/src/main/java/org/apache/drill/exec/planner/logical/partition/PruneScanRule.java [hunk 1]
```diff
@@ -387,16 +378,35 @@ protected void doOnMatch(RelOptRuleCall call, Filter filterRel, Project projectR
...
+              // stop at the first index with no match and use the immediate
+              // previous index
+              index = j-1;
+              break;
+            }
+          }
+          for (int j=0; j <= index; j++) {
+            path += "/" + spInfo[j];
+          }
+          cacheFileRoot = descriptor.getBaseTableLocation() + path;
```
Comment (jinfengni) [inline exec/java-exec/src/main/java/org/apache/drill/exec/planner/logical/partition/PruneScanRule.java]: cacheFileRoot is set within 'IF' branch. Are we going to get a null for cacheFileRoot, if matchBitSet has no bit set? Will cacehFileRoot=null cause issue in downstream logic?

Comment (amansinha100) [inline exec/java-exec/src/main/java/org/apache/drill/exec/planner/logical/partition/PruneScanRule.java]: Actually, cacheFileRoot can be null and ParquetGroupScan will default to using the selectionRoot as the location of the cache file..here's where it decides what to use: 
https://github.com/apache/drill/blob/master/exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/ParquetGroupScan.java#L221

**Comment by jinfengni** [comment]
+1
LGTM.

**Pull request closed without merging**